MobiFlight/Joysticks/Winwing/WinwingCdu.cs

```diff
@@ -93,5 +97,121 @@ public override void Shutdown()
+        {
+            try
+            {
+                // Grab the message
+                string message = string.Format(UserMessageCodes.CodeToMessageMap[messageCode], parameters).ToUpper();
+
+                // Create a list of word tokens
+                string[] tokens = message.Split(' ');
+
+                List<string> lines = new List<string>();
+
+                // Add first token in separate line
+                lines.Add(tokens[0]);
+                lines.Add(string.Empty); // empty line
+                var currentLine = new StringBuilder();
+
+                for (int i = 1; i < tokens.Length; i++)
+                {
+                    string token = tokens[i];
+                    token = token.Length > LINE_LENGTH ? token.Substring(0, LINE_LENGTH) : token;
+
+                    // Does not fit
+                    if (currentLine.Length + token.Length + 1 > LINE_LENGTH)
+                    {
+                        lines.Add(currentLine.ToString());
+                        currentLine.Clear();
+                    }
+
+                    // Append
+                    token = currentLine.Length == 0 ? token : $" {token}";
+                    currentLine.Append(token);
+
+                    // Last token
+                    if (i == tokens.Length - 1)
+                    {
+                        lines.Add(currentLine.ToString());
+                    }
+                }
+                SendToDisplay(lines);
+            }
+            catch (Exception ex)
+            {
+                Log.Instance.log($"WinwingCdu - Error on show user message: {ex.Message}", LogSeverity.Error);
+            }
+        }
+
+        
+        /// <summary>
+        /// Sends a list of text lines to the display, formatting them to fit within the display's dimensions.  
+        /// </summary>
+        /// <remarks>This method formats the provided text lines to fit within the display's fixed
+        /// dimensions, ensuring proper alignment and padding. If the number of lines exceeds the display's capacity,
+        /// only the first 14 lines are used. Empty lines are added to center the text vertically, and each line is
+        /// padded to align horizontally. The formatted data is then converted to a JSON structure and sent to the
+        /// display.</remarks>
+        /// <param name="lines">A collection of strings representing the lines of text to be displayed. Each line is centered horizontally
+        /// and the entire set of lines is centered vertically within the display.</param>
+        private void SendToDisplay(IList<string> lines)
+        {
+            // Center vertically          
+            int emptyLineCount = (14 - Math.Min(lines.Count, 14)) / 2;
+            var linesToDisplay = Enumerable.Repeat(string.Empty, emptyLineCount).Concat(lines).ToList();          
+
+            var cduData = new CduChar[336]; // 14 * 24
```

Comment:
I appreciate I'm just a user and not a member of the review group so please feel free to tell me to get lost if you disagree.

Regarding the Columns, Rows and Cell numbers, is it worth using constants for these to add consistent naming and meaning behind the values?

For example:
```csharp
public static class CduConstants
{
    public const int COLUMNS = 24;
    public const int ROWS= 14;
    public const int CELLS = COLUMNS * ROWS;
}
```

That then provides reusable values and less need for comments

```
int emptyLineCount = (CduConstants.ROWS - Math.Min(lines.Count, CduConstants.ROWS)) / 2;
var cduData = new CduChar[CduConstants.CELLS];
```